tests/services/test_legalhold.py

```diff
@@ -123,7 +124,7 @@ def test_get_all_matter_custodians_calls_get_expected_number_of_times(
             mock_get_all_matter_custodians_response,
             mock_get_all_matter_custodians_empty_response,
         ]
-        for _ in service.get_all_matter_custodians():
+        for _ in service.get_all_matter_custodians(user="test"):
```

Comment:
for some reason I am still not convinced how this works, when we call `get_all_matter_custodians` without any optional argument, it passes but individually calling `get_custodians_page` fails, now if `get_custodians_page` needs one of the optional field so `get_all_matter_custodians` too need to have same behaviour.

Reply:
I mean it passes in the integration tests, and here the change has been done on `get_all_matter_custodians` method instead of `get_custodians_page`.

Reply:
The `user="test"` param is no longer strictly necessary in this test, as initially I had been raising an exception when all of the criteria were `None` (before any http requests were made), so here we needed to pass in _some_ option. 

I haven't looked at the integration test yet, but running `next(sdk.legalhold.get_all_matter_custodians())` in my test org raises the 400 BadRequestError (which we've now replaced with the custom `Py42LegalHoldCriteriaMissingError`.

Reply:
Ah, your test does pass in two arguments: 

```
def test_get_all_matter_custodians(connection, policy, matter):
     response_gen = connection.legalhold.get_all_matter_custodians(policy, matter)
     for response in response_gen:
         assert response.status_code == 200
         break
```

Which if they're valid input values, will do a search for those Ids. It returns a 200, but no results:

```
next(sdk.legalhold.get_all_matter_custodians("1", "1"))
<Py42Response [status=200, data={'legalHoldMemberships': []}]>
```